app/grandchallenge/reader_studies/forms.py

```diff
@@ -486,68 +485,92 @@ class SelectUploadWidget(Select):
-                            attrs={
-                                "reader_study_slug": instance.reader_study.slug,
-                                "display_set_pk": instance.pk,
-                                "interface_pk": interface.pk,
-                            }
-                        ),
-                    )
-
-            self.fields["order"] = IntegerField(initial=instance.order)
+        self.instance = instance
+        self.reader_study = reader_study
+        self.user = user
+        for slug, values in reader_study.values_for_interfaces.items():
+            current_value = None
+            if instance:
+                current_value = instance.values.filter(
+                    interface__slug=slug
+                ).first()
+            interface = ComponentInterface.objects.get(slug=slug)
+            if interface.is_file_kind:
+                self.fields[slug] = self._get_file_field(
+                    interface, values, current_value
+                )
+            elif interface.is_json_kind:
+                self.fields[slug] = self._get_json_field(
+                    interface, current_value
+                )
+            else:
+                self.fields[slug] = self._get_image_field(
+                    interface, values, current_value
+                )
+
+            order = (
+                instance.order
+                if instance
+                else reader_study.next_display_set_order
+            )
+            self.fields["order"] = IntegerField(initial=order)
+
+    def _get_json_field(self, interface, current_value):
+        return InterfaceFormField(
+            kind=interface.kind,
+            schema=interface.schema,
+            initial=current_value.value if current_value else None,
+            use_file_widget=interface.is_file_kind,
+            required=False,
+            user=self.user,
+        ).field
+
+    def _get_image_field(self, interface, values, current_value):
+        return ModelChoiceField(
+            queryset=ComponentInterfaceValue.objects.filter(id__in=values),
+            initial=current_value,
+            required=False,
+            widget=SelectUploadWidget(
+                attrs={
+                    "reader_study_slug": self.reader_study.slug,
+                    "display_set_pk": self.instance.pk,
+                    "interface_pk": interface.pk,
+                }
+            ),
+        )
+
+    def _get_file_field(self, interface, values, current_value):
+        return self._get_image_field(interface, values, current_value)
+
+
+class DisplaySetUpdateForm(DisplaySetBaseForm):
+    def __init__(self, *args, instance, user, **kwargs):
+        super().__init__(
+            *args,
+            instance=instance,
+            reader_study=instance.reader_study,
+            user=user,
+            **kwargs,
+        )
+
+
+class DisplaySetCreateForm(DisplaySetBaseForm):
+    def __init__(self, *args, instance=None, reader_study, **kwargs):
```

Comment:
Why are you not explicitly passing the `user` in your call to super? It's included in the kwargs, so this works either way, but maybe it would be clearer to do this explicitly, just like you do for the UpdateForm?